openmdao/recorders/recording_manager.py

```diff
@@ -190,7 +192,12 @@ def record_viewer_data(problem):
     # if any recorders were found, get the viewer data and record it
     if recorders:
         from openmdao.visualization.n2_viewer.n2_viewer import _get_viewer_data
-        viewer_data = _get_viewer_data(problem)
+        try:
+            viewer_data = _get_viewer_data(problem)
+        except TypeError as err:
+            viewer_data = {}
+            issue_warning(str(err))
+
         viewer_data['md5_hash'] = problem.model._generate_md5_hash()
```

Comment:
If a TypeError is caught above, won't this just raise a KeyError?

Reply:
Not sure I understand your thought here..
If there is a TypeError, the dict is initialized to empty and the 'md5_hash' key is added to that empty dict
(There is a test for this case btw: https://github.com/swryan/OpenMDAO/blob/2684_model_not_group/openmdao/recorders/tests/test_sqlite_reader.py#L1298)